src/store/annotations/reducers.ts

```diff
@@ -1,6 +1,23 @@
+import {CREATE_ANNOTATION} from './actions';
 
-const initialState = [];
+const initialState = {
+  data: [],
+};
+
+function maxId(state) {
+  return Math.max(0, ...state.data.map(annotation => annotation.annotationId)) + 1;
+}
 
 export default function editor(state = initialState, action) {
```

Comment:
As you've named the array field `data`, I'd suggest renaming this function to `annotations`